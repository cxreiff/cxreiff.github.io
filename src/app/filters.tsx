import { FC } from 'react'

export const Filters: FC = () => {
    return (
        <>
            <Turbulence />
        </>
    )
}

const Turbulence = () => (
    <svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="0" height="0">
        <defs>
            <filter id="turbulence_0">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="0" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_1">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="1" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_2">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="2" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_3">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="3" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_4">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="4" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_5">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="5" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_6">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="6" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
            <filter id="turbulence_7">
                <feTurbulence baseFrequency="0.02" numOctaves="3" result="noise" seed="7" />
                <feDisplacementMap in="SourceGraphic" in2="noise" scale="3" />
            </filter>
        </defs> 
    </svg>
)
