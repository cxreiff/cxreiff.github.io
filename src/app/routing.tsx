import { FC, ReactNode, Suspense, useEffect, lazy } from "react";
import { Routes, Route, useLocation } from "react-router-dom";

import { Loader } from "src/common/loader/loader.component";
import { IFrame } from "src/common/iframe/iframe.component";
import {
  DOCUMENT_TITLE_ROOT,
  BREAKOUT_URL,
  PLATFORMER_URL,
  HEX_TILES_URL,
  PIXIJS_URL,
  BABYLONJS_URL,
  PLAIN_WEBGL_URL,
  BIGFOOT_MAP_URL,
} from "src/utilities/constants";

import Homepage from "src/features/homepage/homepage.component";
import Projects from "src/features/projects/projects.component";
import Photos from "src/features/photos/photos.component";
import Sketches from "src/features/sketches/sketches.component";
import ProjectsFigrid from "src/features/projects/projectsFigrid.component";

// const Posts = lazy(() => import("src/features/posts/posts.component"));
// const PostsContent = lazy(
//   () => import("src/features/posts/postsContent.component")
// );

const Shaders = lazy(() => import("src/features/shaders/shaders.component"));

const Pixijs = () => (
  <IFrame aspect={15.5 / 9} url={PIXIJS_URL} title={"pixijs"} canFullscreen />
);
const Babylonjs = () => (
  <IFrame
    aspect={3 / 2}
    url={BABYLONJS_URL}
    title={"babylonjs"}
    canFullscreen
  />
);
const Breakout = () => (
  <IFrame
    aspect={3 / 2}
    mobileAspect={2 / 3}
    url={BREAKOUT_URL}
    title={"breakout"}
    canFullscreen
  />
);
const Platformer = () => (
  <IFrame
    aspect={16 / 10}
    url={PLATFORMER_URL}
    title={"platformer"}
    canFullscreen
  />
);
const HexTiles = () => (
  <IFrame
    aspect={16 / 10}
    url={HEX_TILES_URL}
    title={"hex tiles"}
    canFullscreen
  />
);
const PlainWebGL = () => (
  <IFrame
    aspect={16 / 10}
    url={PLAIN_WEBGL_URL}
    title={"hex tiles"}
    canFullscreen
  />
);
const BigfootMap = () => (
  <IFrame aspect={16 / 10} url={BIGFOOT_MAP_URL} title={"bigfoot sightings"} />
);

export type RouteDefinition = {
  path: string;
  Component: FC;
  children?: RouteDefinition[];
};

export const ROUTES: RouteDefinition[] = [
  { path: "", Component: Homepage },
  {
    path: "projects",
    Component: Projects,
    children: [
      {
        path: "figrid",
        Component: ProjectsFigrid,
      },
      {
        path: "pixijs",
        Component: Pixijs,
        children: [{ path: ":id", Component: Pixijs }],
      },
      { path: "babylonjs", Component: Babylonjs },
      { path: "breakout", Component: Breakout },
      { path: "platformer", Component: Platformer },
      { path: "hex_tiles", Component: HexTiles },
      { path: "webgl", Component: PlainWebGL },
      { path: "bigfoot", Component: BigfootMap },
      {
        path: "shaders",
        Component: Shaders,
        children: [{ path: ":category", Component: Shaders }],
      },
      {
        path: "sketches",
        Component: Sketches,
        children: [{ path: ":id", Component: Sketches }],
      },
    ],
  },
  { path: "photos", Component: Photos },
  // {
  //   path: "posts",
  //   Component: Posts,
  //   children: [{ path: ":id", Component: PostsContent }],
  // },
];

export const Routing = () => {
  const { pathname } = useLocation();

  useEffect(() => {
    const baseRoute = pathname.split("/")[1];
    document.title = baseRoute
      ? `${DOCUMENT_TITLE_ROOT} â€” ${baseRoute}`
      : DOCUMENT_TITLE_ROOT;
  }, [pathname]);

  const renderRoutes = (routes: RouteDefinition[], prefix = "") =>
    routes.reduce((renderedRoutes, { path, Component, children }) => {
      const fullPath = `${prefix}/${path}`;

      if (children) {
        renderedRoutes.push(...renderRoutes(children, fullPath));
      }

      return renderedRoutes.concat(
        <Route path={fullPath} element={<Component />} />
      );
    }, [] as ReactNode[]);

  return (
    <Suspense fallback={<Loader />}>
      <Routes>{renderRoutes(ROUTES)}</Routes>
    </Suspense>
  );
};
