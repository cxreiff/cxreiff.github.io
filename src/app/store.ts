import { createStore, applyMiddleware, compose } from 'redux'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { connectRouter, routerMiddleware } from 'connected-react-router'

import history from './history'
import homepage from '~/src/features/homepage/homepage.slice'

const router = connectRouter(history)

const middleware = [
    ...getDefaultMiddleware(),
    routerMiddleware(history),
];
 
if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger');
  middleware.push(logger);
}

const store = configureStore({
    reducer: {
        router,
        homepage,
    },
    middleware,
})

export type AppDispatch = typeof store.dispatch
export const useAppDispatch = () => useDispatch<AppDispatch>()

export type RootState = ReturnType<typeof store.getState>
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

export default store

