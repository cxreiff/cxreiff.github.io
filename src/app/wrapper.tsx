import React, { ComponentType, ReactElement } from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'connected-react-router'

import store from './store'
import history from './history'

const wrapper = (Component: ComponentType, props = {}): ReactElement => {
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Component {...props} />
      </ConnectedRouter>
    </Provider>
  )
}

export default wrapper
