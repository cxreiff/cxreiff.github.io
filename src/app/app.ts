import { FC } from 'react'
import 'normalize.css'

import wrapper from './wrapper'
import Layout from '~/src/features/layout/layout.component'

const App: FC = () => wrapper(Layout)

export default App
