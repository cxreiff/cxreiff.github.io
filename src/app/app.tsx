import { FC } from 'react'

import { Layout } from '~/src/features/layout/layout.component'
import { Wrapper } from './wrapper'
import { Filters } from './filters'
import { Routing, ROUTES } from './routing'

const App: FC = () => {
    return (
        <Wrapper>
            <Layout routes={ROUTES}>
                <Routing />
            </Layout>
            <Filters />
        </Wrapper>
    )
}

export default App
