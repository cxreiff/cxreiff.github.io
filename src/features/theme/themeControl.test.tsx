import { render, fireEvent } from '@testing-library/preact'

import { Wrapper } from '~/src/app/wrapper'
import { ThemeControl } from './themeControl.component'

import styles from './themeControl.module.scss'

test('should render', () => {
    const { container, getAllByRole } = render(<ThemeControl />, { wrapper: Wrapper })

    expect(getAllByRole('button')).toHaveLength(3)

    expect(container).toMatchSnapshot()
})

test('should swap the current theme', () => {
    const { getByRole } = render(<ThemeControl />, { wrapper: Wrapper })

    const blueButton = getByRole('button', { name: 'change theme to blue color scheme' })
    fireEvent.click(getByRole('button', { name: 'change theme to blue color scheme' }))

    expect(blueButton).toHaveClass(styles.current_theme)
})
