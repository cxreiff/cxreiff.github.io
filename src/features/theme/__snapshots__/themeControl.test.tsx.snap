// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <span>
    <button
      aria-label="change theme to red color scheme"
      class="lightmode_button current_theme"
    />
    <button
      aria-label="change theme to green color scheme"
      class="greenish_button"
    />
    <button
      aria-label="change theme to blue color scheme"
      class="bluish_button"
    />
  </span>
</div>
`;
